Add lumping_interval Feature for Semian Circuit Breakers (#606)
Shopify/semian: README.md

There are four configuration parameters for circuit breakers in Semian:

- **circuit_breaker**. Enable or Disable Circuit Breaker. Defaults to `true` if not set.
- **error_threshold**. The amount of errors a worker encounters within `error_threshold_timeout`
  amount of time before opening the circuit,
  that is to start rejecting requests instantly.
- **error_threshold_timeout**. The amount of time in seconds that `error_threshold`
  errors must occur to open the circuit.
  Defaults to `error_timeout` seconds if not set.
- **error_timeout**. The amount of time in seconds until trying to query the resource
  again.
- **error_threshold_timeout_enabled**. If set to false it will disable
  the time window for evicting old exceptions. `error_timeout` is still used and
  will reset the circuit. Defaults to `true` if not set.
- **success_threshold**. The amount of successes on the circuit until closing it
  again, that is to start accepting all requests to the circuit.
- **half_open_resource_timeout**. Timeout for the resource in seconds when
  the circuit is half-open (supported for MySQL, Net::HTTP and Redis).

> - **lumping_interval**. Interval for timeframe of errors to be lumped and recorded as one.

Reviewer: ```suggestion
- **lumping_interval**. If provided, errors within this interval will be lumped and recorded as one.
```

Reviewer 2: would also be good to mention that it's in seconds.
